src/Git/Git.h

```diff
@@ -35,11 +35,32 @@ struct git_repository;
 class CFilterData
 {
 public:
+
+	enum
+	{
+		SHOW_LAST_N_COMMITS = 0,
+		SHOW_LAST_SELECTED_DATE = 1,
+		SHOW_LAST_N_YEARS = 2,
+		SHOW_LAST_N_MONTHS = 3,
+		SHOW_LAST_N_WEEKS = 4,
+	};
+
 	CFilterData()
 	{
 		m_From=m_To=-1;
 		m_IsRegex=1;
+		m_NumberOfLogsScale = (DWORD)CRegDWORD(_T("Software\\TortoiseGit\\NumberOfLogsScale"), -1);
+		m_NumberOfLogs = (DWORD)CRegDWORD(_T("Software\\TortoiseGit\\NumberOfLogs"), 0);
+		m_NumberOfYears = (DWORD)CRegDWORD(_T("Software\\TortoiseGit\\NumberOfYears"), 0);
+		m_NumberOfMonths = (DWORD)CRegDWORD(_T("Software\\TortoiseGit\\NumberOfMonths"), 0);
+		m_NumberOfWeeks = (DWORD)CRegDWORD(_T("Software\\TortoiseGit\\NumberOfWeeks"), 0);

```

Comment:
I think we need.
1. Can switch different range of time quickly in Log Message dialog. (1 Year, 6 months(half year), 13 weeks(3 months)).

2. For first scale - say last commits, the "number of" setting value could be 10000. If we just have one "number of" setting, once  user switches to "year" scale, it will be 10000 year, user will need to modify that setting value. mum... not bad, not good.

3. The bad thing I could count is too many code added. :P

P.S. "number of" setting value for "last selected date".